examples/federated-learning/tff/nvflare/README.md

```diff
@@ -0,0 +1,296 @@
+   NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID="${NVFLARE_EXAMPLE_CONTAINER_IMAGE_REPOSITORY_HOSTNAME}/$(terraform output -raw container_image_repository_name)/nvflare-tensorflow"
+   NVFLARE_EXAMPLE_CONTAINER_IMAGE_BUILD_CONTEXT_PATH="../examples/federated-learning/tff/nvflare/container-image"
+   NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID_WITH_TAG=${NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID}:${NVFLARE_EXAMPLE_CONTAINER_IMAGE_TAG}
+   export NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID
+   export NVFLARE_EXAMPLE_CONTAINER_IMAGE_TAG="0.0.1"
+
+   docker build \
+     --file "${NVFLARE_EXAMPLE_CONTAINER_IMAGE_BUILD_CONTEXT_PATH}/Dockerfile" \
+     --tag "${NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID_WITH_TAG}" \
+     ${NVFLARE_EXAMPLE_CONTAINER_IMAGE_BUILD_CONTEXT_PATH}
+   ```
+
+1. Authenticate Docker with the Artifact Registry repository:
+
+   ```bash
+   gcloud auth configure-docker \
+       "${NVFLARE_EXAMPLE_CONTAINER_IMAGE_REPOSITORY_HOSTNAME}"
+   ```
+
+1. Push the container image to the Artifact Registry repository:
+
+   ```bash
+   docker image push "${NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID_WITH_TAG}"
+   ```
+
+1. Create the ConfigSync folder that will deploy the workload
+
+   ```bash
+   ACM_CONFIG_SYNC_PATH=$(terraform output -raw acm_config_sync_configuration_destination_directory_path)
+   mkdir -p ${ACM_CONFIG_SYNC_PATH}/nvflare/base-storage
+   ```
+
+1. Modify the `kustomization.yaml.tpl` file to add the new name and new tag of the image:
+
+   ```bash
+   envsubst < ../examples/federated-learning/tff/nvflare/templates/kustomization.yaml.tpl > ${ACM_CONFIG_SYNC_PATH}/kustomization.yaml
+   ```
+
+### Create infrastructure to store the models
+
+All the models generated will be stored in a Cloud storage bucket mounted by each pod.
+
+1. Initialize the following Terraform variables:
+
+   ```hcl
+   nvflare = true
+   ```
+
+1. Deploy the infrastructure for NVFlare
+
+   ```bash
+   terraform apply
+   ```
+
+1. Modify the `pv-workspace.yaml.tpl` file to replace the bucket name:
+
+   ```bash
+   NVFLARE_WORKSPACE_BUCKET_NAME="\"$(terraform output -raw nvflare_workspace_bucket_name)\""
+   export NVFLARE_WORKSPACE_BUCKET_NAME
+   envsubst < ../examples/federated-learning/tff/nvflare/templates/pv-workspace.yaml.tpl > ${ACM_CONFIG_SYNC_PATH}/base-storage/pv-workspace.yaml
+   ```
+
+1. Regenerate the `kustomization.yaml` file with all the necessary informations:
+
+   ```bash
+   NVFLARE_EXAMPLE_WORKLOADS_KUBERNETES_NAMESPACE="$(terraform output -raw nvflare_workloads_kubernetes_namespace)"
+   export NVFLARE_EXAMPLE_WORKLOADS_KUBERNETES_NAMESPACE
+   envsubst < ../examples/federated-learning/tff/nvflare/templates/pv-workspace.yaml.tpl > ${ACM_CONFIG_SYNC_PATH}/base-storage/pv-workspace.yaml
+   ```
+
+### Create NVFlare folder structure
+
+Now that the persistent volume is created, you can start creating the folder structure and upload it to Cloud Storage.
+
+1. Install nvflare on your workstation:
```

Comment:
At this point, we're still in the `terraform` directory. From where shall we run this?